service/src/main/java/org/dice_research/fc/sparql/query/QueryEngineCustomHTTP.java

```diff
@@ -0,0 +1,257 @@
+        throw new UnsupportedOperationException("Invalid operation");
+    }
+
+    @Override
+    public Query getQuery() {
+        return this.query;
+    }
+
+    @Override
+    public ResultSet execSelect() {
+        String result = createRequest();
+        ResultSetFactory fac = new ResultSetFactory();
+
+        // the result is not a valid XML then replace with an empty XML
+        if(result.length()<10) {
+            result = emptyXML();
+        }
+
+        ResultSet resultSet = fac.fromXML(result);
+
+        return resultSet;
+    }
+
+    /**
+     * the empty xml used to generate empty ResultSet
+     * @return string which is an empty xml
+     */
+    private String emptyXML() {
+        return "<sparql xmlns=\"http://www.w3.org/2005/sparql-results#\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:schemaLocation=\"http://www.w3.org/2001/sw/DataAccess/rf1/result2.xsd\"><head></head><results distinct=\"false\" ordered=\"true\"></results></sparql>>";
+    }
+
+    /**
+     * read an InputStream and return the String
+     * @param content is an InputStream
+     * @return string which is a content of input stream
+     */
+    private String read(InputStream content) {
+        ByteSource byteSource = new ByteSource() {
+            @Override
+            public InputStream openStream() throws IOException {
+                return content;
+            }
+        };
+
+        try {
+            return byteSource.asCharSource(Charsets.UTF_8).read();
+        } catch (IOException e) {
+            LOGGER.error("Could not read stream due to ",e);
+        }
+        return "";
+    }
+
+
+    /**
+     * run the query and return the result
+     * when the timeout reached the query terminated and should handle in catch
+     * @return string which is a result of the query
+     */
+    private String createRequest() {
+        RequestConfig config = RequestConfig.custom()
+                .setConnectTimeout(timeout)
+                .setConnectionRequestTimeout(timeout)
+                .setSocketTimeout(timeout).build();
+        try(CloseableHttpClient client = HttpClientBuilder.create().setDefaultRequestConfig(config).build()){
+            HttpGet get = new HttpGet(service + "?query=" + URLEncoder.encode(query.toString(), "UTF-8"));
+            get.addHeader(HttpHeaders.ACCEPT, "application/sparql-results+xml");
+            HttpResponse resp = client.execute(get);
+            String responseContent = read(resp.getEntity().getContent());
+            InputStream is = new ByteArrayInputStream(responseContent.getBytes(StandardCharsets.UTF_8));
+            String result = IOUtils.toString(is, StandardCharsets.UTF_8);
+            LOGGER.info(result);
+            return result;
+        }
+        catch(java.net.SocketTimeoutException e) {
+            LOGGER.info("Timeout this query: "+query.toString());
+            return "";
+        }
+        catch(Exception e){
+            e.printStackTrace();
```

Comment:
You can remove printing the stack trace. You throw the exception anyway. So no need to print it.